src/main/java/com/jcraft/jsch/KeyPair.java

```diff
@@ -930,8 +936,19 @@ else if (data != null &&
             if(buf[0]=='s'&& buf[1]=='s'&& buf[2]=='h' && buf[3]=='-'){
               if(prvkey==null &&
                  buf.length>7){
-                if(buf[4]=='d'){ type=DSA; }
-                else if(buf[4]=='r'){ type=RSA; }
+                  if (buf[4] == 'd') {
+                    if (buf.length >= 12 && buf[8] == 'c' && buf[9] == 'e' && buf[10] == 'r' && buf[11] == 't') {
+                        type = DSA_CERT;
+                    } else {
+                        type = DSA;
+                    }
+                  } else if (buf[4] == 'r') {
+                    if (buf.length >= 12 && buf[8] == 'c' && buf[9] == 'e' && buf[10] == 'r' && buf[11] == 't') {
+                        type = RSA_CERT;
+                    } else {
+                        type = RSA;
+                    }
+                }
```

Comment:
It would be nice to maintain the existing coding style here (namely no spaces surrounding `=`, `==`, etc.).